modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -236,10 +239,24 @@ class PodSpecBuilder {
             volumeClaims.addAll( opts.getVolumeClaims() )
         // -- labels
         if( opts.labels ) {
-            def keys = opts.labels.keySet()
-            if( 'app' in keys ) throw new IllegalArgumentException("Invalid pod label -- `app` is a reserved label")
-            if( 'runName' in keys ) throw new IllegalArgumentException("Invalid pod label -- `runName` is a reserved label")
-            labels.putAll( opts.labels )
+            for( Map.Entry<String,String> entry : opts.labels ) {
+                if( !entry.key || !entry.value ) {
+                    log.debug "K8s invalid label entry=$entry"  
+                }
+                else if( entry.key == 'app' ){
+                     throw new IllegalArgumentException("Invalid pod label -- `app` is a reserved label")
+                }
+                else if( entry.key == 'runName' ){
+                     throw new IllegalArgumentException("Invalid pod label -- `runName` is a reserved label")
+                }
+                else if( entry.value.size() > 63 ) {
+                    log.debug "K8s label exceeds allowed size: 63 -- offending entry=$entry"  
+                    labels.put(entry.key, entry.value.substring(0,63))   
+                }
+                else {
+                    labels.put(entry.key, entry.value)    
+                }
+            } 
         }
```

Comment:
Make sense to separate this snippet into its own method other this method too complex, something like 

```
Map<String, String> normalizeLabels(Map<String, String> labels) {
 // the same as above
}
```

The it becomes:

```
if( opts.labels ) {
  labels.putAll(normaliseLabels(opts.label)) 
}